docs/lab/lab04/lab04.md

```diff
@@ -0,0 +1,384 @@
+:::
+
+### PWM in Embassy-rs
+
+First, we need a reference to all peripherals, as usual. 
+
+```rust
+// Initialize peripherals
+let peripherals = embassy_rp::init(Default::default());
+```
+
+In order to modify the PWM counter configurations, we need to create a `Config` for our PWM.
+
+```rust
+// Create config for PWM slice
+let mut config: Config = Default::default();
+// Set top value (value at which PWM counter will reset)
+config.top = 0x8000; // in HEX, equals 32768 in decimal
+// Set compare value (counter value at which the PWM signal will change from 1 to 0)
+config.compare_a = config.top / 2;
+```
+
+In the example above:
+    - `top` is the field from `Config` that will define the value at which the counter will reset back to 0
+    - `compare_a` is the field from `Config` that will define the value at which the PWM signal will switch from 1 to 0
+In this case, `compare_a` is half of `config.top`. This means that the duty cycle of the generated PWM signal will be 50%, or, in other words, that the PWM signal will switch from 1 to 0 halfway through each period.
+
+To select the pin that we want to use for PWM, we need to create a new PWM driver that uses the correct channel and output for our pin.
+
+```rust
+// Create PWM driver for pin 0
+let mut pwm = Pwm::new_output_a( // output A
+    peripherals.PWM_CH0, // channel 0
+    peripherals.PIN_0, // pin 0
+    config.clone()
+);
+```
+
+:::warning
+1. The code above is an example for pin 0. You need to modify the channel, output and pin depending on the PWM pin you choose to use!
+2. The value of `compare_a` or `compare_b` must be changed depending on the desired duty cycle!
+:::
+
+If we decide to modify the value of `compare_a` or `compare_b`, we have to update the configuration for the PWM.
+
+```rust
+config.compare_a += 100; // modified value of `compare_a`
+pwm.set_config(&config); // set the new configuration for PWM
+```
+
+## Analog-to-Digital Converter (ADC)
+
+Now we know how to represent an analog signal using digital signals. There are plenty of cases in which we need to know how to transform an analog signal into a digital one, for example a temperature reading, or the voice of a person. This means that we need to correctly represent a continuous wave of infinite values to a discrete wave of a finite set of values.
+For this, we need to sample the analog signal periodically, in other words to measure the analog signal at a fixed interval of time. This is done by using an **Analog-to-Digital converter**.
+
+The *sampling rate* is the frequency at which a new sample is read. The higher the sampling rate, the more samples we get, so the more accurate the representation of the signal.
+
+The *resolution* is the number of bits which we can use in order to store the value of the sample. The higher the resolution, the more values we can store, so the more accurate the representation.
```

Comment:
What about using a table?

The ADC has two important parameters ...

Reply:
Done.